### DISCUSSION TITLE
Postgres to Planetscale migration with UUID v4s

### DISCUSSION THREAD

**Discussion by Cristianooo** [planetscale/discussion#106] [Q&A] [answered]
I am trying to migrate from postgres to planetscale but the biggest gripe I have is MySQL's lack of support for a uuid column type. 
Here's an example of ddl for a table in postgres. Notice the uuid data type and uuid default generator.
```
create table "agentWebsites"(
	"websiteId" serial primary key,
	title varchar(50),
	theme varchar(20),
	"websiteUrl" uuid default uuid_generate_v4(),
	"agentId" int
        "domainId" int
)
````

I use [slugid](https://github.com/taskcluster/slugid) which uses uuid v4, so I cant simply change the column to a char(36) and call MySQL's built in uuid() function because that creates v1 uuids. 

I found some [MySQL functions](https://stackoverflow.com/questions/32965743/how-to-generate-a-uuidv4-in-mysql) that create v4 uuids but I'm having trouble implementing them for planetscale. 

Whenever I try to add a function, I get a simple syntax error-
![error](https://user-images.githubusercontent.com/22159514/134562916-7d05d4c2-568c-471a-bf21-065ce1f59a3e.JPG)

**Comment by jgreet** [comment, marked as answer]
Unfortunately PlanetScale doesn't support CREATE FUNCTION or CREATE PROCEDURE yet. When those features are [working in Vitess](https://vitess.io/docs/reference/compatibility/mysql-compatibility/#stored-procedures) we'll bring them over to PlanetScale but I can't make any promises at this point.

The built in UUID functions you mentioned will work on PlanetScale, but since you need v4 UUIDs I don't think there's any option other than to generate them in your application instead of in the database.

**Reply by Cristianooo** [reply]
Yeah I thought I might have to do that.  Just didnt want to because thats the most work 😬 haha. But ok thanks for your help again :)
